### PR TITLE
Add Numeric Support to Dialect

### LINKED ISSUE #198: Add support for usage of BigDecimal
After the NUMERIC feature was added to Spanner, BigDecimal is now a supported type to use in Hibernate entities.

Please add support for this feature.

### PR DESCRIPTION
**Pull request by dzou**
Adds Numeric support to the dialect. Fixes #198

Seems we really just have to do this in the `SpannerDialect`:

```
registerColumnType(Types.DECIMAL, "NUMERIC");
registerColumnType(Types.NUMERIC, "NUMERIC");
```

cc/ @olavloite

### CODE REVIEW

**Review by dmitry-s — DISMISSED** [review]
LGTM

**Comment by dmitry-s** [comment]
@dzou do we need to update docs?

**Review by meltsufin — DISMISSED** [review]
Looks good, unless you agree that we should update the last paragraph of the README as part of this PR. :)

File: google-cloud-spanner-hibernate-samples/basic-hibernate-sample/src/main/java/com/example/Payment.java [hunk 1]
```diff
@@ -40,7 +41,7 @@
   @Type(type = "uuid-char")
   private UUID id;
 
-  private Long amount;
+  private BigDecimal amount;
```
Comment (elefeint) [inline google-cloud-spanner-hibernate-samples/basic-hibernate-sample/src/main/java/com/example/Payment.java:44]: Are we still testing `Long` somewhere else?

**Review by elefeint — APPROVED** [review]
LGTM, just one question about whether we are losing `Long` coverage.

File: google-cloud-spanner-hibernate-samples/basic-hibernate-sample/src/main/java/com/example/Payment.java (hunk 1 above)
Comment (dzou) [reply google-cloud-spanner-hibernate-samples/basic-hibernate-sample/src/main/java/com/example/Payment.java]: No worries, we have plenty of `Long` coverage (among other data types) in the Hibernate integration tests that we copied from the Hibernate repo: https://github.com/GoogleCloudPlatform/google-cloud-spanner-hibernate/blob/master/google-cloud-spanner-hibernate-testing/src/main/java/org/hibernate/userguide/model/Payment.java#L28

**Review by elefeint — APPROVED** [review]

**Pull request merged**